hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/hudi/HoodieComparator.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hudi
+
+class HoodieComparator[S] extends Ordering[S] {
+  import com.google.common.collect.Ordering
```

Comment:
we cannot use guava inside Hudi. Its an illegal import in java code. Love to again understand why this is needed.

Reply:
`JavaRDD.scala` implements sortBy , which internally uses `sortBy` of `RDD.scala`
JavaRDD ->uses  implicit natural ordering 
`def sortBy[S](f: JFunction[T, S], ascending: Boolean, numPartitions: Int): JavaRDD[T] = {
    def fn: (T) => S = (x: T) => f.call(x)
    import com.google.common.collect.Ordering  // shadows scala.math.Ordering
    implicit val ordering = Ordering.natural().asInstanceOf[Ordering[S]]
    implicit val ctag: ClassTag[S] = fakeClassTag
    wrapRDD(rdd.sortBy(fn, ascending, numPartitions))
  }`
 to keep behaviour consistent I have used the same natural ordering for non-nullable objects .

Reply:
Great job digging deeper. :). Love to avoid the custom RDD approach if possible as mentioned above

Reply:
@vinothchandar  have made it string sort (should do null_first)